src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -52,34 +72,131 @@ export async function initialize({
             return await onLocalInvokeCommand({
                 commandName: command,
                 lambdaLocalInvokeParams: params,
+                configuration,
+                toolkitOutputChannel,
+                processInvoker,
+                taskInvoker,
+                telemetryService
             })
         },
     })
 }
 
 /**
  * The command that is run when user clicks on Run Local or Debug Local CodeLens
+ * Accepts object containing the following params:
+ * @param configuration - SettingsConfiguration (for invokeLambdaFunction)
+ * @param toolkitOutputChannel - "AWS Toolkit" output channel
  * @param commandName - Name of the VS Code Command currently running
  * @param lambdaLocalInvokeParams - Information about the Lambda Handler to invoke locally
+ * @param processInvoker - SAM CLI Process invoker
+ * @param taskInvoker - SAM CLI Task invoker
+ * @param telemetryService - Telemetry service for metrics
  */
 async function onLocalInvokeCommand({
+    configuration,
+    toolkitOutputChannel,
     commandName,
     lambdaLocalInvokeParams,
+    processInvoker,
+    taskInvoker,
+    telemetryService
 }: {
+    configuration: SettingsConfiguration
+    toolkitOutputChannel: vscode.OutputChannel,
     commandName: string,
     lambdaLocalInvokeParams: LambdaLocalInvokeParams,
+    processInvoker: SamCliProcessInvoker,
+    taskInvoker: SamCliTaskInvoker,
+    telemetryService: TelemetryService
 }): Promise<{ datum: Datum }> {
 
+    const channelLogger = getChannelLogger(toolkitOutputChannel)
     const runtime = await getRuntimeForLambda({
```

Comment:
Looks like getRuntimeForLambda can throw Errors. Should it go in the try block?

Reply:
We're not try-catching this in the other languages. At the very least, we need it outside the block to capture the runtime for the getMetricDatum return at the end. Thoughts on remedying this would be welcome.

Reply:
Good point. It looks like the other languages make this call outside of the function and pass in the runtime (which would satisfy the metricDatum). I'm okay considering this out of scope.